tests/mir-opt/README.md

```diff
@@ -49,3 +49,21 @@ This exists mainly for completeness and is rarely useful.
 ```
 // EMIT_MIR $file_name_of_some_mir_dump.before.mir
 ```
+
+# FileCheck directives
+
+The LLVM FileCheck tool is used to verify the contents of output MIR against `CHECK` directives
+present in the test file. This works on the runtime MIR, generated by `--emit=mir`, and not
+on the output of a individual passes.
+
+Use `// skip-filecheck` to prevent FileCheck from running.
+
+To check MIR for function `foo`, start with a `// CHECK-LABEL fn foo(` directive.
+
+`{{regex}}` syntax allows to match `regex`.
+
+`[[name:regex]]` syntax allows to bind `name` to a string matching `regex`, and refer to it
+as `[[name]]` in later directives, `regex` should be written not to match a leading space.
+Use `[[my_local:_.*]]` to name a local, and `[[my_bb:bb.*]]` to name a block.
```

Comment:
Would `[[my_local:_[0-9]+]]` be better to avoid matching random other things that start with `_`? Same for basic blocks.

Reply:
That would be more precise yes. However, I don't think there is any risk of matching the wrong thing. Basic-blocks can only appear in very few places. Likewise, I don't know anything that could be mistaken for a local.